### PR TITLE
Setup minimal CI

### PR DESCRIPTION
**Pull request by rnburn**
Sets up a CI enviornment (modeled after envoy https://github.com/envoyproxy/envoy/tree/master/ci) and adds basic test targets for bazel and cmake on linux. 

See ci/README.md for usage instructions.

This will, of course, need to expand to include tests for formatting, OS X, Windows, different compilers, ASAN, TSAN, etc.

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+if [[ "$1" == "cmake.test" ]]; then
+  cd "${BUILD_DIR}"
+  cmake -DCMAKE_BUILD_TYPE=Debug  \
+        -DCMAKE_CXX_FLAGS="-Werror" \
+        "${SRC_DIR}"
+  make
+  make test
+  exit 0
+elif [[ "$1" == "bazel.test" ]]; then
+  bazel build -c dbg $BAZEL_OPTIONS -- //...
```
Comment (g-easy) [inline ci/do_ci.sh]: Why dbg?

File: ci/setup_ci_environment.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                autogen \
+                autoconf \
+                libtool \
+                gnupg2 \
+                ssh \
+                wget \
+                libgtest-dev
+
+# Follows these instructions for setting up gtest
+# https://www.eriksmistad.no/getting-started-with-google-test-on-ubuntu/
```
Comment (g-easy) [inline ci/setup_ci_environment.sh]: Webpage says "Note that this package only install source files. You have to compile the code yourself to create the necessary library files." but `dpkg -L libgtest-dev` says

```
/usr/lib/x86_64-linux-gnu/libgtest.a
/usr/lib/x86_64-linux-gnu/libgtest_main.a
```

File: ci/setup_ci_environment.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+set -e
+apt-get update 
+apt-get install --no-install-recommends --no-install-suggests -y \
+                build-essential \
+                cmake \
+                pkg-config \
+                git \
+                ca-certificates \
+                curl \
+                automake \
```
Comment (g-easy) [inline ci/setup_ci_environment.sh]: We can probably reduce this list. I don't think we need autotools.

File: ci/setup_ci_environment.sh (hunk 2 above)
Comment (rnburn) [reply ci/setup_ci_environment.sh]: I tried it first without that extra step and cmake couldn't find libgtest

File: ci/do_ci.sh (hunk 1 above)
Comment (rnburn) [reply ci/do_ci.sh]: I changed to the default

File: ci/setup_ci_environment.sh (hunk 3 above)
Comment (rnburn) [reply ci/setup_ci_environment.sh]: removed the extra packages

File: ci/setup_ci_environment.sh (hunk 2 above)
Comment (g-easy) [reply ci/setup_ci_environment.sh]: Odd. Idk what's going on here so let's keep it as-is for now. :)

File: ci/run_docker.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -e
+
+BUILD_IMAGE=opentelemetry-cpp-build3
```
Comment (g-easy) [inline ci/run_docker.sh]: 3?

Comment (rnburn) [reply ci/run_docker.sh]: Fixed

**Pull request merged**